ppr-ui/src/utils/utilities.ts

```diff
@@ -107,7 +117,11 @@ export const deepChangesComparison = (base: BaseDataUnionIF, current: BaseDataUn
     const keys1 = Object.keys(base)
     const keys2 = Object.keys(current)
 
-    return keys1.length !== keys2.length || keys1.some(key => deepChangesComparison(base[key], current[key]))
+    if (keys1.length !== keys2.length) {
+      return true
+    }
+    return keys1.some(key => deepChangesComparison(base[key], current[key], isCaseSensitive))
```

Comment:
Why expand the simplified statement into long form with an `if` condition and double returns? 

`return keys1.length !== keys2.length || keys1.some(key => deepChangesComparison(base[key], current[key], isCaseSensitive))` Is the exact same with less code and more concise no?

Reply:
I was getting `...reaching memory limit: JS heap out of memory` error when running the tests. This fixed it, because it does not need to evaluate the second condition (and won't go deeper into recursion).

![Screenshot 2024-03-04 at 16 49 42](https://github.com/bcgov/ppr/assets/2333290/89049f28-ae2a-4851-9a9c-aef64ca943bb)

Reply:
This doesn't seem right then, there was no issue before. 

Please look a bit deeper into this, changing code syntax to pass unit tests is somewhat of a red flag, especially considering the simplicity of the statement.

Reply:
Let me know if you want me to run the code and help with the unit test debugging.

Reply:
Yea, thanks, I spent few hours today trying to figure it out. But I have another solution, so will try it. Will let you know.

Reply:
Also, the build tests didn't have the same js heap issue.

Reply:
Ok, fair enough, if you've already spent time looking into it than all good. 
It does seem related to test set up, which is why i lean to leaving the code as it was and refactoring the test setup but i'll leave it to you to decide.

Reply:
I restarted my machine, cleared cache, reinstalled the node, etc. and the tests passed, once... Now it is failing again. Very strange.

Reply:
Regarding this, I tried to revert to original logic, but it failed with memory error.

Reply:
Hmm yea, as you've noted in your other PR it's not really related to this specific logic, so i think it's safe to revert and continue to debug, or rerun the unit test suite until it passes since it's intermittent. 

We can focus on this as a single issue and not let it block this PR but we can now be pretty certain it's not related to the || statement, as your now seeing this in all the PR's. 
My recommendation is to revert and continue to run the tests until they pass and we can tackle this issue standalone, i don't think it's related to any code in this PR.

Reply:
Yea I agree it's not probably related to this. However, I still think that we could avoid logical OR, just because we don't want to run the second conditional statement (`keys1.some(key => deepChangesComparison...`) if first condition fails. It might be fast compare, but unnecessary?

Reply:
Updated as per your recommendation.